### PR TITLE
8345355: [s390x] support for z16 hardware

### PR DESCRIPTION
**Pull request by offamitkumar**
This was the output for `jdk/bin/java -XX:+Verbose --version` command on a z16 hardware: 

```
CPU Version as detected internally: system-z, g9-z15, ldisp_fast, extimm, pcrel_load/store, cmpb, cond_load/store, interlocked_update, txm, vectorinstr, instrext2, venh1, instrext3, venh2, out-of-support_as_of_tbd, aes128, aes192, aes256, sha1, sha256, sha512, ghash
```

So the issue is that z16 should be recognized as g10-z16 but instead I got g9-z15. 

This is the output with current changes: 

```
CPU Version as detected internally: system-z, g10-z16, ldisp_fast, extimm, pcrel_load/store, cmpb, cond_load/store, interlocked_update, txm, vectorinstr, instrext2, venh1, instrext3, venh2,bear_enh, sort_enh, nnpa_assist, storage_key_removal, vpack_decimal_enh, out-of-support_as_of_tbd, aes128, aes192, aes256, sha1, sha256, sha512, ghash
```

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8345355](https://bugs.openjdk.org/browse/JDK-8345355): [s390x] support for z16 hardware (**Enhancement** - P4)

### Reviewers
 * [Lutz Schmidt](https://openjdk.org/census#lucy) (@RealLucy - **Reviewer**)
 * [Andrew Haley](https://openjdk.org/census#aph) (@theRealAph - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22508/head:pull/22508` \
`$ git checkout pull/22508`

Update a local copy of the PR: \
`$ git checkout pull/22508` \
`$ git pull https://git.openjdk.org/jdk.git pull/22508/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22508`

View PR using the GUI difftool: \
`$ git pr show -t 22508`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22508.diff">https://git.openjdk.org/jdk/pull/22508.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22508#issuecomment-2513640772)
</details>

### CODE REVIEW

File: src/hotspot/cpu/s390/vm_version_s390.hpp [hunk 1]
```diff
@@ -88,6 +88,9 @@ class VM_Version: public Abstract_VM_Version {
 // z13 end
 #define  MiscInstrExt2Mask              0x0000000000000020UL  // z14
 #define  MiscInstrExt3Mask              0x0000000000000004UL  // z15
+
+#define BEAREnhFacilityMask             0x4000000000000000UL  // z16, BEAR-enhancement facility, Bit: 193
+#define NNPAssistFacilityMask           0x0000000000040000UL  // z16, Neural-network-processing-assist facility, Bit: 165
```
Comment (offamitkumar) [inline src/hotspot/cpu/s390/vm_version_s390.hpp:93]: [My script](https://github.com/offamitkumar/s390xFacilityMask) says that this should be 
```suggestion
#define NNPAssistFacilityMask           0x0000000004000000UL  // z16, Neural-network-processing-assist facility, Bit: 165
```

```bash
Enter Facility Bit Number: 165
_feature array index: 2
bit position to set: 37
bit string: 0000000000000000000000000000000000000100000000000000000000000000
Mask:  0x0000000004000000UL
```

**Comment by offamitkumar** [comment]
@RealLucy 👀

**Review by RealLucy — APPROVED** [review]
LGTM.
It's good to have new H/W detected correctly.

**Review by theRealAph — APPROVED** [review]

**Comment by offamitkumar** [comment]
Thanks for the approval Lutz, Andrew :-) 

/integrate

**Pull request closed without merging**